src/roswire/common/source.py

```diff
@@ -29,100 +32,70 @@ class SourceLanguage(enum.Enum):
     PYTHON = "python"
 
 
-@attr.s(frozen=True, auto_attribs=True, slots=True)
-class NodeSourceInfo:
-    node_name: str
+class ExecutableKind(enum.Enum):
+    NODE = "node"
+    LIBRARY = "library"
+
+
+@attr.s(auto_attribs=True, slots=True)
+class ExecutableInfo:
+    name: t.Optional[str]
```

Comment:
We probably need to think about the difference between the target name and the target's relative install destination.